[Glemen Neo] iP

glemenneo opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is FREE!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

glemenneo committed:
Level-1

glemenneo committed:
closed_scanner_for_Level-1_plus_done_Level-2

glemenneo committed:
Level-3

glemenneo committed:
Level-4

glemenneo committed:
Level-5

glemenneo committed:
Level-6

glemenneo committed:
A-TextUiTesting

glemenneo committed:
Level-7

glemenneo committed:
Level-8

glemenneo committed:
Merge remote-tracking branch 'origin/branch-Level-8'

Merge branch-Level-8

glemenneo committed:
A-MoreOOP

glemenneo committed:
A-Packages

glemenneo committed:
A-Packages

glemenneo committed:
A-Gradle

glemenneo committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into HEAD

glemenneo committed:
A-JUnit

glemenneo committed:
A-JUnit

glemenneo committed:
Fix merge conflicts

glemenneo committed:
A-JavaDoc

glemenneo committed:
A-CodingStandard

glemenneo committed:
Level-9

glemenneo committed:
Merge_branch-A-CodingStandard

glemenneo committed:
Merge branch 'branch-Level-9'

bigcrushes commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,92 @@
+            if (!input.contains(" at ")) {
+                throw new DukeException("\t☹ OOPS!!! Formatting of event is incorrect.");
+            }
+            String[] split = input.replace("event", "").split(" at ");
```
While this works, an alternative to explore is input.split(" ", 2) at the start to separate the event command and splitting again with /at after. It might work better for events that have the word event in them!

bigcrushes commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+package duke.tasklist;
```
Good separation of files into different packages as necessary.

bigcrushes commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+        this.list = new ArrayList<>();
+    }
+
+    public void addTask(Task task) {
```
Adding some javadocs to some of this functions might be useful

bigcrushes reviewed (commented):
Overall code looks good and largely follows coding standards.

autumn-sonata commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,92 @@
+        } else if (input.contains("todo")) {
+            String content = input.replace("todo", "").trim();
+            if (content.length() == 0) {
+                throw new DukeException("\t☹ OOPS!!! The description of a todo cannot be empty.");
```
Instead of multiple `DukeException` objects with similar error messages, consider making a child class of `DukeException` to handle similar exception issues.

autumn-sonata commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+            System.out.println("\tWelcome new user!");
+        }
+
+        boolean quit = false;
```
Boolean naming should follow the Java coding standard.
```suggestion
    boolean needsToQuit = false;
```

autumn-sonata commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,20 @@
+public class ParserTest {
+    @Test
+    public void byeTest() {
+        Boolean result = false;
```
Small nit, `Boolean` refers to the Object type boolean while `boolean` refers to the primitive type. Consider using the primitive type instead. 👍

autumn-sonata commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+
+public class TaskList {
+
+    private ArrayList<Task> list;
```
Try to be more descriptive with the attribute naming. I know that it is intuitive that this list under the TaskList refers to the list of tasks, but it is just good practice to have descriptive naming.

autumn-sonata commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+import duke.parser.Parser;
+import duke.storage.Storage;
+
+import java.io.IOException;
```
I think this should be placed with line 5, `import java.io.FileNotFoundException`.

autumn-sonata reviewed (commented):
Good job on following most of the coding standards! 👍 Only a few nit-picks.

glemenneo committed:
Add GFMD in PR description

glemenneo committed:
Add CheckStyle

glemenneo committed:
GUI

glemenneo committed:
Improve code quality and readability

Parsing method named Parser.FeedDuke() and parsing method using
contains().

Edge cases are present where string inputs parsed using contains()
would wrongly recognise task descriptions as commands.

Let's rename Parser.FeedDuke() to Parser.parseInput() and for
Parser.parseInput(), string inputs will be parsed using .equals()
and .split().

Also changed and declared some variable names to make code more
understandable.

Adjusted import statements to be lexographically correct.

Many instances of Boolean data type instead of primitive boolean data
type used.

Inefficient code as auto boxing and unboxing will need to be done.

Let's change all instances of Boolean data type to primitive boolean data
type.

glemenneo committed:
Improve class file organisations and updat parsing methods to match Main

glemenneo committed:
Refactor code to improve code quality and readability

Class files are all in duke folder.
Very messy in terms of package structuring.
Move all class files into their own sub packages.
Cleaner code and package organisation.
In line with main branch package organisation.

Parser.java parsing logic is all in one parseInput() method.
Method body is very long and there is a lot of repeated knowledge.
Create new methods parseTask() and parseIndexCommands() to handle
different task types.
Improves code organisation and and applies DRY principle.
Also added enums and made use of switch cases to improve code
readability.

parseInput command: correct split to check for "at" instead of "by".
parseInput command checks for "at" instead of "by" to extract date and
time.

This is not inline with the user input specifications where for a event
command, the date and time are added after the "at" keyword.

Let's
- change the split() command to check for "at" instead of "by" keyword.

glemenneo committed:
Merge branch 'branch-Level-10'

Merge branch-Level-10 into master

glemenneo committed:
Add assertions for methods that change list size and parsing date-time methods

glemenneo committed:
Improve code quality by adding JavaDocs and splitting long lines of
code

glemenneo committed:
Merge pull request #1 from glemenneo/branch-A-Assertions

Merge branch-A-Assertions into master

glemenneo committed:
Merge branch 'master' into branch-A-CodeQuality

Merge master into branch-A-CodeQuality

glemenneo committed:
Merge pull request #2 from glemenneo/branch-A-CodeQuality

Merge branch-A-CodeQuality into master

glemenneo committed:
Add extension C-Tagging

glemenneo committed:
Merge branch 'master' of https://github.com/glemenneo/ip

Merge pull from remote with C-Tagging changes

glemenneo committed:
Fix unresolved merge conflicts

glemenneo committed:
Add JavaDocs and improved code quality

glemenneo committed:
Add Ui picture and user guide

glemenneo committed:
Change build.gradle to prepare for creating jar file

glemenneo committed:
Change storage to create directories and be system independent

glemenneo committed:
Fixed javaFx dependencies

glemenneo committed:
Change Parser class method to non static

glemenneo committed:
Change task saving to occur after every valid command

glemenneo committed:
Change Event and Deadline seperators to include /
